orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/servergroup/strategies/MonitoredDeployStrategy.groovy

```diff
@@ -74,18 +75,28 @@ class MonitoredDeployStrategy implements Strategy {
   DeploymentMonitorServiceProvider deploymentMonitorServiceProvider
 
   @Override
-  List<Stage> composeBeforeStages(Stage parent) {
-    return composeFlow(parent)
-      .findAll({ it.syntheticStageOwner == SyntheticStageOwner.STAGE_BEFORE })
-  }
+  List<Stage> composeBeforeStages(Stage stage) {
```

Comment:
this is annoying, but some strategies (like RRB) took advantage of being called during `plan` and mutating the current stage. Hence this code lives in `composeBeforeStages` which is also called during stage `plan`ning